CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 2.7)
+project(R-Host)
+
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+    message(FATAL_ERROR "In-source builds are not allowed. Please invoke CMake from ./build, or use ./build.sh helper script.")
+endif()
+
+if(CMAKE_BUILD_TYPE MATCHES DEBUG)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_HOME_DIRECTORY}/bin/Release)
```

Comment:
Our .vcxproj is going to invoke CMake and make from MSYS2. I'm still working on that, and it's not a part of this PR. This one just lets you build it manually from MSYS2 shell.

(Note that this is a PR into another branch - the one Karthik created for cross-plat work - not into master.)